packages/material-ui-system/src/styleFunctionSx/extendSxProp.js

```diff
@@ -0,0 +1,30 @@
+import { propToStyleFunction } from '../getThemeValue';
+
+const splitProps = (props) => {
+  const result = {
+    systemProps: {},
+    otherProps: {},
+  };
+
+  Object.keys(props).forEach((prop) => {
+    if (propToStyleFunction[prop]) {
+      result.systemProps[prop] = props[prop];
+    } else {
+      result.otherProps[prop] = props[prop];
+    }
+  });
+
+  return result;
+};
+
+export default function extendSxProp(props) {
+  const { sx: inSx = {}, ...other } = props;
```

Comment:
We can spread undefined, it won't create any issue.

```suggestion
  const { sx: inSx, ...other } = props;
```